modules/post/windows/gather/enum_application_artifacts_packrat.rb

```diff
@@ -0,0 +1,918 @@
+##
+# This module requires Metasploit: http//metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'msf/core/post/windows/user_profiles'
+
+class Metasploit3 < Msf::Post
+
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info={})
+    super( update_info( info,
+    'Name'         => 'Windows Gather Application Artifacts (PackRat)',
+    'Description'  => %q{
+     PackRat gathers artifacts of various categories from a large number of applications.
+
+     Artifacts include: chat logins and logs, browser logins and history and cookies,
+     email logins and emails sent and received and deleted, contacts, and many others.
+     These artifacts are collected from applications including:
+     12 browsers, 13 chat/IM/IRC applications, 6 email clients, and 1 game.
+
+     The use case for this post-exploitation module is to specify the types of
+     artifacts you are interested in, to gather the relevant files depending on your aims.
+
+     Please refer to the options for a full list of filter categories.
+    },
+    'License'      => MSF_LICENSE,
+    'Author'       => [
+      'Barwar Salim M',  # Leeds Beckett University student
+      'Z. Cliffe Schreuders (http://z.cliffe.schreuders.org)'  # Leeds Beckett University lecturer
+    ],
+    'Platform'     => %w{win},
+    'SessionTypes' => ['meterpreter']
+    ))
+
+    register_options(
+    [
+      OptBool.new('STORE_LOOT', [false, 'Store artifacts into loot database (otherwise, only download)', 'true']),
+      # enumerates the options based on the artifacts that are defined below
+      OptEnum.new('APPCATEGORY', [false, 'Category of applications to gather from', 'All', @@apps.map{ |x| x[:category] }.uniq.unshift('All')]),
+      OptEnum.new('APPLICATION', [false, 'Specify application to gather from', 'All', @@apps.map{ |x| x[:application] }.uniq.unshift('All')]),
+      OptEnum.new('ARTIFACTS', [false, 'Type of artifacts to collect', 'All', @@apps.map{ |x| x[:filetypes] }.uniq.unshift('All')]),
+], self.class)
+  end
+
+  # this associative array defines the artifacts known to PackRat
+  @@apps= [
```

Comment:
I don't get why you're using a class variable here, unless there is a reason I'm forgetting right now this array looks like should be a constant